# Guide project

In this guided project, we'll work with a dataset of used cars from eBay Kleinanzeigen, a classifieds section of the German eBay website. The aim is to clean the dataset and perform some initial analysis on it.

In [1]:
#Import libraries and read csv
import pandas as pd
import numpy as np

autos = pd.read_csv('autos.csv', encoding = 'latin1')

In [2]:
autos

dateCrawled                                               name  \
0      2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1      2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2      2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3      2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4      2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
...                    ...                                                ...   
49995  2016-03-27 14:38:19   Audi_Q5_3.0_TDI_qu._S_tr.__Navi__Panorama__Xenon   
49996  2016-03-28 10:50:25  Opel_Astra_F_Cabrio_Bertone_Edition___TÜV_neu+...   
49997  2016-04-02 14:44:48                     Fiat_500_C_1.2_Dualogic_Lounge   
49998  2016-03-08 19:25:42                 Audi_A3_2.0_TDI_Sportback_Ambition   
49999  2016-03-14 00:42:12                                Opel_Vectra_1.6_16V   

       seller offerType    price   abtest vehicleType  yearOfRegistration  \
0      privat   Angebot   $5,000  control         bus                2004   
1      privat   Angebot   $8,500  control   limousine                1997   
2      privat   Angebot   $8,990     test   limousine                2009   
3      privat   Angebot   $4,350  control  kleinwagen                2007   
4      privat   Angebot   $1,350     test       kombi                2003   
...       ...       ...      ...      ...         ...                 ...   
49995  privat   Angebot  $24,900  control   limousine                2011   
49996  privat   Angebot   $1,980  control      cabrio                1996   
49997  privat   Angebot  $13,200     test      cabrio                2014   
49998  privat   Angebot  $22,900  control       kombi                2013   
49999  privat   Angebot   $1,250  control   limousine                1996   

         gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0        manuell      158  andere  150,000km                    3      lpg   
1      automatik      286     7er  150,000km                    6   benzin   
2        manuell      102    golf   70,000km                    7   benzin   
3      automatik       71  fortwo   70,000km                    6   benzin   
4        manuell        0   focus  150,000km                    7   benzin   
...          ...      ...     ...        ...                  ...      ...   
49995  automatik      239      q5  100,000km                    1   diesel   
49996    manuell       75   astra  150,000km                    5   benzin   
49997  automatik       69     500    5,000km                   11   benzin   
49998    manuell      150      a3   40,000km                   11   diesel   
49999    manuell      101  vectra  150,000km                    1   benzin   

            brand notRepairedDamage          dateCreated  nrOfPictures  \
0         peugeot              nein  2016-03-26 00:00:00             0   
1             bmw              nein  2016-04-04 00:00:00             0   
2      volkswagen              nein  2016-03-26 00:00:00             0   
3           smart              nein  2016-03-12 00:00:00             0   
4            ford              nein  2016-04-01 00:00:00             0   
...           ...               ...                  ...           ...   
49995        audi              nein  2016-03-27 00:00:00             0   
49996        opel              nein  2016-03-28 00:00:00             0   
49997        fiat              nein  2016-04-02 00:00:00             0   
49998        audi              nein  2016-03-08 00:00:00             0   
49999        opel              nein  2016-03-13 00:00:00             0   

       postalCode             lastSeen  
0           79588  2016-04-06 06:45:54  
1           71034  2016-04-06 14:45:08  
2           35394  2016-04-06 20:15:37  
3           33729  2016-03-15 03:16:28  
4           39218  2016-04-01 14:38:50  
...           ...                  ...  
4999

In [3]:

autos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   dateCrawled          50000 non-null  object
 1   name                 50000 non-null  object
 2   seller               50000 non-null  object
 3   offerType            50000 non-null  object
 4   price                50000 non-null  object
 5   abtest               50000 non-null  object
 6   vehicleType          44905 non-null  object
 7   yearOfRegistration   50000 non-null  int64 
 8   gearbox              47320 non-null  object
 9   powerPS              50000 non-null  int64 
 10  model                47242 non-null  object
 11  odometer             50000 non-null  object
 12  monthOfRegistration  50000 non-null  int64 
 13  fuelType             45518 non-null  object
 14  brand                50000 non-null  object
 15  notRepairedDamage    40171 non-null  object
 16  date

In [4]:
autos.head()

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   
4  privat   Angebot  $1,350     test       kombi                2003   

     gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0    manuell      158  andere  150,000km                    3      lpg   
1  automatik      286     7er  150,000km                    6   benzin   
2    manuell      102    golf   70,000km                    7   benzin   
3  automatik       71  fortwo   70,000km                    6   benzin   
4    manuell        0   focus  150,000km                    7   benzin   

        brand notRepairedDamage          dateCreated  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

We can see there are a several null in some columns, like vehicle type... Because there are 50000 rows, and .info() show us less in these columns
The dataset contains 20 columns, most of which are strings.
Some columns have null values, but none have more than ~20% null values.

In [5]:
autos.columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [6]:
autos.columns = ['date_crawled', 'name', 'seller', 'offer_type', 'price', 'abtest',
       'vehicle_type', 'registration_year', 'gearbox', 'power_ps', 'model',
       'odometer', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damage', 'ad_created', 'nr_of_pictures', 'postal_code',
       'last_seen']

In [7]:
autos

date_crawled                                               name  \
0      2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1      2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2      2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3      2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4      2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
...                    ...                                                ...   
49995  2016-03-27 14:38:19   Audi_Q5_3.0_TDI_qu._S_tr.__Navi__Panorama__Xenon   
49996  2016-03-28 10:50:25  Opel_Astra_F_Cabrio_Bertone_Edition___TÜV_neu+...   
49997  2016-04-02 14:44:48                     Fiat_500_C_1.2_Dualogic_Lounge   
49998  2016-03-08 19:25:42                 Audi_A3_2.0_TDI_Sportback_Ambition   
49999  2016-03-14 00:42:12                                Opel_Vectra_1.6_16V   

       seller offer_type    price   abtest vehicle_type  registration_year  \
0      privat    Angebot   $5,000  control          bus               2004   
1      privat    Angebot   $8,500  control    limousine               1997   
2      privat    Angebot   $8,990     test    limousine               2009   
3      privat    Angebot   $4,350  control   kleinwagen               2007   
4      privat    Angebot   $1,350     test        kombi               2003   
...       ...        ...      ...      ...          ...                ...   
49995  privat    Angebot  $24,900  control    limousine               2011   
49996  privat    Angebot   $1,980  control       cabrio               1996   
49997  privat    Angebot  $13,200     test       cabrio               2014   
49998  privat    Angebot  $22,900  control        kombi               2013   
49999  privat    Angebot   $1,250  control    limousine               1996   

         gearbox  power_ps   model   odometer  registration_month fuel_type  \
0        manuell       158  andere  150,000km                   3       lpg   
1      automatik       286     7er  150,000km                   6    benzin   
2        manuell       102    golf   70,000km                   7    benzin   
3      automatik        71  fortwo   70,000km                   6    benzin   
4        manuell         0   focus  150,000km                   7    benzin   
...          ...       ...     ...        ...                 ...       ...   
49995  automatik       239      q5  100,000km                   1    diesel   
49996    manuell        75   astra  150,000km                   5    benzin   
49997  automatik        69     500    5,000km                  11    benzin   
49998    manuell       150      a3   40,000km                  11    diesel   
49999    manuell       101  vectra  150,000km                   1    benzin   

            brand unrepaired_damage           ad_created  nr_of_pictures  \
0         peugeot              nein  2016-03-26 00:00:00               0   
1             bmw              nein  2016-04-04 00:00:00               0   
2      volkswagen              nein  2016-03-26 00:00:00               0   
3           smart              nein  2016-03-12 00:00:00               0   
4            ford              nein  2016-04-01 00:00:00               0   
...           ...               ...                  ...             ...   
49995        audi              nein  2016-03-27 00:00:00               0   
49996        opel              nein  2016-03-28 00:00:00               0   
49997        fiat              nein  2016-04-02 00:00:00               0   
49998        audi              nein  2016-03-08 00:00:00               0   
49999        opel              nein  2016-03-13 00:00:00               0   

       postal_code            last_seen  
0            79588  2016-04-06 06:45:54  
1            71034  2016-04-06 14:45:08  
2            35394  2016-04-06 20:15:37  
3            33729  2016-03-15 03:16:28  
4            39218  2016-04-01 1

We change the columns name from 'camel Case' to 'snake_case', because is more confort for work and friendly for python. Also, we did the next changes: 
yearOfRegistration to registration_year
monthOfRegistration to registration_month
notRepairedDamage to unrepaired_damage
dateCreated to ad_created

In [8]:
autos.describe(include = 'all')

date_crawled         name  seller offer_type  price abtest  \
count                 50000        50000   50000      50000  50000  50000   
unique                48213        38754       2          2   2357      2   
top     2016-04-02 11:37:04  Ford_Fiesta  privat    Angebot     $0   test   
freq                      3           78   49999      49999   1421  25756   
mean                    NaN          NaN     NaN        NaN    NaN    NaN   
std                     NaN          NaN     NaN        NaN    NaN    NaN   
min                     NaN          NaN     NaN        NaN    NaN    NaN   
25%                     NaN          NaN     NaN        NaN    NaN    NaN   
50%                     NaN          NaN     NaN        NaN    NaN    NaN   
75%                     NaN          NaN     NaN        NaN    NaN    NaN   
max                     NaN          NaN     NaN        NaN    NaN    NaN   

       vehicle_type  registration_year  gearbox      power_ps  model  \
count         44905       50000.000000    47320  50000.000000  47242   
unique            8                NaN        2           NaN    245   
top       limousine                NaN  manuell           NaN   golf   
freq          12859                NaN    36993           NaN   4024   
mean            NaN        2005.073280      NaN    116.355920    NaN   
std             NaN         105.712813      NaN    209.216627    NaN   
min             NaN        1000.000000      NaN      0.000000    NaN   
25%             NaN        1999.000000      NaN     70.000000    NaN   
50%             NaN        2003.000000      NaN    105.000000    NaN   
75%             NaN        2008.000000      NaN    150.000000    NaN   
max             NaN        9999.000000      NaN  17700.000000    NaN   

         odometer  registration_month fuel_type       brand unrepaired_damage  \
count       50000        50000.000000     45518       50000             40171   
unique         13                 NaN         7          40                 2   
top     150,000km                 NaN    benzin  volkswagen              nein   
freq        32424                 NaN     30107       10687             35232   
mean          NaN            5.723360       NaN         NaN               NaN   
std           NaN            3.711984       NaN         NaN               NaN   
min           NaN            0.000000       NaN         NaN               NaN   
25%           NaN            3.000000       NaN         NaN               NaN   
50%           NaN            6.000000       NaN         NaN               NaN   
75%           NaN            9.000000       NaN         NaN               NaN   
max           NaN           12.000000       NaN         NaN               NaN   

                 ad_created  nr_of_pictures   postal_code            last_seen  
count                 50000         50000.0  50000.000000                50000  
unique                   76             NaN           NaN                39481  
top     2016-04-03 00:00:00             NaN           NaN  2016-04-07 06:17:27  
freq                   1946             NaN           NaN                    8  
mean                    NaN             0.0  50813.627300                  NaN  
std                     NaN             0.0  25779.747957                  NaN  
min                     NaN             0.0   1067.000000                  NaN  
25%                     NaN             0.0  30451.000000                  NaN  
50%                     NaN             0.0  49577.000000                  NaN  
75%                     NaN             0.0  71540.000000                  NaN  
max                     NaN             0.0  99998.000000                  NaN

Cells with wrong informatio: seller, offer_type, nr_of_pictures
price and odometer are numeric values stored as text

In [9]:
#Replace $ and ',' for '' in price
autos['price'] = autos['price'].str.replace('$','', regex=True)
autos['price'] = autos['price'].str.replace(',','', regex=True)
autos['price'] = autos['price'].astype(int)

In [10]:
#Replace km and ',' for '' in odometer
autos['odometer'] = autos['odometer'].str.replace('km','', regex=True)
autos['odometer'] = autos['odometer'].str.replace(',','', regex=True)
autos['odometer'] = autos['odometer'].astype(int)

In [11]:
#rename odometer column to odometer_km

autos.rename({'odometer': 'odometer_km'}, axis=1, inplace=True)

autos.head()

date_crawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offer_type  price   abtest vehicle_type  registration_year  \
0  privat    Angebot   5000  control          bus               2004   
1  privat    Angebot   8500  control    limousine               1997   
2  privat    Angebot   8990     test    limousine               2009   
3  privat    Angebot   4350  control   kleinwagen               2007   
4  privat    Angebot   1350     test        kombi               2003   

     gearbox  power_ps   model  odometer_km  registration_month fuel_type  \
0    manuell       158  andere       150000                   3       lpg   
1  automatik       286     7er       150000                   6    benzin   
2    manuell       102    golf        70000                   7    benzin   
3  automatik        71  fortwo        70000                   6    benzin   
4    manuell         0   focus       150000                   7    benzin   

        brand unrepaired_damage           ad_created  nr_of_pictures  \
0     peugeot              nein  2016-03-26 00:00:00               0   
1         bmw              nein  2016-04-04 00:00:00               0   
2  volkswagen              nein  2016-03-26 00:00:00               0   
3       smart              nein  2016-03-12 00:00:00               0   
4        ford              nein  2016-04-01 00:00:00               0   

   postal_code            last_seen  
0        79588  2016-04-06 06:45:54  
1        71034  2016-04-06 14:45:08  
2        35394  2016-04-06 20:15:37  
3        33729  2016-03-15 03:16:28  
4        39218  2016-04-01 14:38:50

In [12]:
autos['odometer_km'].unique().shape

(13,)

In [13]:
autos['odometer_km'].describe()

count     50000.000000
mean     125732.700000
std       40042.211706
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64

In [14]:
autos['odometer_km'].value_counts().head()

150000    32424
125000     5170
100000     2169
90000      1757
80000      1436
Name: odometer_km, dtype: int64

In [15]:
autos['price'].unique().shape

(2357,)

In [16]:
autos['price'].describe()

count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64

In [17]:
autos['price'].sort_values(ascending = True)

18089           0
43923           0
15225           0
29499           0
8445            0
           ...   
27371    12345678
47598    12345678
39377    12345678
42221    27322222
39705    99999999
Name: price, Length: 50000, dtype: int64

In [18]:
autos = autos[(autos['price'] > 0) & (autos['price'] <= 351000)]
print(len(autos))



48565


We removed the rows with prices above 1000000 and below 1, commonly prices like '0', '12345678' or '11111111' are write to fill the space, no because it is the real price.

In [19]:
print(autos['date_crawled'].str[:10].value_counts(normalize=True, dropna=False).sort_index())

2016-03-05    0.025327
2016-03-06    0.014043
2016-03-07    0.036014
2016-03-08    0.033296
2016-03-09    0.033090
2016-03-10    0.032184
2016-03-11    0.032575
2016-03-12    0.036920
2016-03-13    0.015670
2016-03-14    0.036549
2016-03-15    0.034284
2016-03-16    0.029610
2016-03-17    0.031628
2016-03-18    0.012911
2016-03-19    0.034778
2016-03-20    0.037887
2016-03-21    0.037373
2016-03-22    0.032987
2016-03-23    0.032225
2016-03-24    0.029342
2016-03-25    0.031607
2016-03-26    0.032204
2016-03-27    0.031092
2016-03-28    0.034860
2016-03-29    0.034099
2016-03-30    0.033687
2016-03-31    0.031834
2016-04-01    0.033687
2016-04-02    0.035478
2016-04-03    0.038608
2016-04-04    0.036487
2016-04-05    0.013096
2016-04-06    0.003171
2016-04-07    0.001400
Name: date_crawled, dtype: float64


In [20]:
print(autos['ad_created'].str[:10].value_counts(normalize=True, dropna=False).sort_index())

2015-06-11    0.000021
2015-08-10    0.000021
2015-09-09    0.000021
2015-11-10    0.000021
2015-12-05    0.000021
                ...   
2016-04-03    0.038855
2016-04-04    0.036858
2016-04-05    0.011819
2016-04-06    0.003253
2016-04-07    0.001256
Name: ad_created, Length: 76, dtype: float64


In [21]:
print(autos['last_seen'].str[:10].value_counts(normalize=True, dropna=False).sort_index())

2016-03-05    0.001071
2016-03-06    0.004324
2016-03-07    0.005395
2016-03-08    0.007413
2016-03-09    0.009595
2016-03-10    0.010666
2016-03-11    0.012375
2016-03-12    0.023783
2016-03-13    0.008895
2016-03-14    0.012602
2016-03-15    0.015876
2016-03-16    0.016452
2016-03-17    0.028086
2016-03-18    0.007351
2016-03-19    0.015834
2016-03-20    0.020653
2016-03-21    0.020632
2016-03-22    0.021373
2016-03-23    0.018532
2016-03-24    0.019767
2016-03-25    0.019211
2016-03-26    0.016802
2016-03-27    0.015649
2016-03-28    0.020859
2016-03-29    0.022341
2016-03-30    0.024771
2016-03-31    0.023783
2016-04-01    0.022794
2016-04-02    0.024915
2016-04-03    0.025203
2016-04-04    0.024483
2016-04-05    0.124761
2016-04-06    0.221806
2016-04-07    0.131947
Name: last_seen, dtype: float64


We can see spike in some days in last_seen

In [22]:
autos['registration_year'].describe()

count    48565.000000
mean      2004.755421
std         88.643887
min       1000.000000
25%       1999.000000
50%       2004.000000
75%       2008.000000
max       9999.000000
Name: registration_year, dtype: float64

The max year of registration is 9999, that's an error, max would be a 2024 that is the present year. And the minimun value is 1000, that's another error, maybe could consider after 1886, date when the vehicle was created

In [23]:
autos = autos[(autos['registration_year']>1900) & (autos['registration_year']<=2016)]

We remove the rows with registration year outside (1900-2016) because is the last year where take registers and 1900 is the first year of the last ccentury

In [24]:
autos['registration_year'].value_counts(normalize=True)

2000    0.067608
2005    0.062895
1999    0.062060
2004    0.057904
2003    0.057818
          ...   
1939    0.000021
1948    0.000021
1938    0.000021
1953    0.000021
1943    0.000021
Name: registration_year, Length: 78, dtype: float64

The minority of cars on sale are register in 20 century

In [25]:
top_20 = autos['brand'].value_counts(ascending=False).head(20).index
print(top_20)

Index(['volkswagen', 'bmw', 'opel', 'mercedes_benz', 'audi', 'ford', 'renault',
       'peugeot', 'fiat', 'seat', 'skoda', 'nissan', 'mazda', 'smart',
       'citroen', 'toyota', 'hyundai', 'sonstige_autos', 'volvo', 'mini'],
      dtype='object')


We select the top 20 brands by amount of cars, to explore data

In [26]:
#Mean price per brand
avg_price_brand = {}

for brand in top_20:
    # Make in one line with .loc[mask, index_to_work]
    mean = autos.loc[autos['brand'] == brand, 'price'].mean().round(2)
    #Two line, first mask and results autos filter after mean
    '''sel = autos[autos['brand'] == brand]
    mean = sel['price'].mean()'''
    
    avg_price_brand[brand] = mean
    

print(avg_price_brand)

{'volkswagen': 5402.41, 'bmw': 8332.82, 'opel': 2975.24, 'mercedes_benz': 8628.45, 'audi': 9336.69, 'ford': 3749.47, 'renault': 2474.86, 'peugeot': 3094.02, 'fiat': 2813.75, 'seat': 4397.23, 'skoda': 6368.0, 'nissan': 4743.4, 'mazda': 4112.6, 'smart': 3580.22, 'citroen': 3779.14, 'toyota': 5167.09, 'hyundai': 5365.25, 'sonstige_autos': 12338.55, 'volvo': 4946.5, 'mini': 10613.46}


In [28]:
#Milage mean per brand
avg_milage_brand = {}
for brand in top_20:
    # Make in one line with .loc[mask, index_to_work]
    mean = autos.loc[autos['brand'] == brand, 'odometer_km'].mean().round(2)
    #Two line, first mask and results autos filter after mean
    '''sel = autos[autos['brand'] == brand]
    mean = sel['price'].mean()'''
    
    avg_milage_brand[brand] = mean

print(avg_milage_brand)

{'volkswagen': 128707.16, 'bmw': 132572.51, 'opel': 129310.04, 'mercedes_benz': 130788.36, 'audi': 129157.39, 'ford': 124266.01, 'renault': 128071.33, 'peugeot': 127153.63, 'fiat': 117121.97, 'seat': 121131.3, 'skoda': 110848.56, 'nissan': 118331.0, 'mazda': 124464.03, 'smart': 99326.78, 'citroen': 119694.19, 'toyota': 115944.35, 'hyundai': 106442.31, 'sonstige_autos': 89956.33, 'volvo': 138067.92, 'mini': 88105.13}


In [37]:
#Convert both dictionaries to series objects, using the series constructor.
#Create a dataframe from dictionary. Dictionary->Series->DataFrame
means = pd.DataFrame(pd.Series(avg_price_brand), columns = ['mean_price'])
#Add a new column Dic->Serie->DataFrame
means['mean_milage'] = pd.DataFrame(pd.Series(avg_milage_brand))

means

mean_price  mean_milage
volkswagen         5402.41    128707.16
bmw                8332.82    132572.51
opel               2975.24    129310.04
mercedes_benz      8628.45    130788.36
audi               9336.69    129157.39
ford               3749.47    124266.01
renault            2474.86    128071.33
peugeot            3094.02    127153.63
fiat               2813.75    117121.97
seat               4397.23    121131.30
skoda              6368.00    110848.56
nissan             4743.40    118331.00
mazda              4112.60    124464.03
smart              3580.22     99326.78
citroen            3779.14    119694.19
toyota             5167.09    115944.35
hyundai            5365.25    106442.31
sonstige_autos    12338.55     89956.33
volvo              4946.50    138067.92
mini              10613.46     88105.13

The majority cars brands has more than 100000 kms. The brand with less avg kms is mini